lib/mcb/commands/courses/create.rb

```diff
@@ -0,0 +1,184 @@
+        menu.choices(*(subject_choices + ['no more subjects'])) do |cmd|
+          cmd == 'no more subjects' ? finished = true : @ucas_subjects << @possible_ucas_subjects.detect {|s| s.subject_name == cmd }
+        end
+      end
+    end
+  end
+
+  def ask_sites
+    finished = false
+    until finished
+      @cli.choose do |menu|
+        sites = !@sites.empty? ? "(#{@sites.map(&:location_name).join(", ")} so far)" : ""
+        menu.prompt = "Which training locations to assign?#{sites}  "
+
+        site_choices = @possible_sites.order(:location_name).pluck(:location_name)
+        menu.choices(*(site_choices + ['no more training locations'])) do |cmd|
+          cmd == 'no more training locations' ? finished = true : @sites << @possible_sites.detect {|s| s.location_name == cmd }
+        end
+      end
+    end
+  end
+end
+
+class CoursePreview
+  def initialize(course)
+    @course = course
+  end
+
+  def to_h
+    {
+      "Age range" => @course.age_range,
+      "Course code" => @course.course_code,
+      "Start date" => @course.start_date.strftime("%b %Y"),
+      "Description" => @course.description,
+      "Provider" => @course.provider.provider_code,
+      "Accredited body" => @course&.accrediting_provider.provider_code || "Self-accrediting",
+      "Route" => @course.program_type,
+      "Modular" => @course.modular,
+      "English" => @course.english,
+      "Maths" => @course.maths,
+      "Science" => @course.science,
+      "Subjects" => @course.dfe_subjects.join(", "),
+      "Level" => @course.dfe_level,
+      "Training locations" => @course.sites.map(&:location_name).join(", "),
+    }
+  end
+end
+
+def confirm_creation_of(course)
+  puts "\nAbout to create the following course:"
+  puts Terminal::Table.new rows: CoursePreview.new(course).to_h
+
+  print "Continue? "
+
+  response = $stdin.readline
+  response.match %r{^y(es?)?}i
+end
+
+run do |opts, args, _cmd|
+  MCB.init_rails(opts)
+  wizard = NewCourseWizard.new(HighLine.new)
+
+  Provider.connection.transaction do
+    args.each do |provider_code|
+      provider = Provider.find_by!(provider_code: provider_code)
+
+      wizard.run(
+        provider: provider,
+        possible_ucas_subjects: Subject.all,
+        possible_sites: Site.where(provider: provider),
+      )
+
+      verbose "Creating a new course for provider #{provider_code}"
+      course = Course.new(
+        provider: provider,
+        age_range: wizard.age_range,
+        course_code: wizard.course_code,
+        name: wizard.name,
+        profpost_flag: :postgraduate,
+        program_type: :scitt_programme,
+        qualification: wizard.qualification,
+        start_date: wizard.start_date,
+        study_mode: wizard.study_mode,
+        accrediting_provider: wizard.accredited_body,
+        modular: nil,
+        english: 3,
+        maths: 3,
+        science: nil,
+        subjects: wizard.ucas_subjects,
+        sites: wizard.sites,
+      )
```

Comment:
As a general pattern, would it maybe make sense to construct a `Course` object inside of `NewCourseWizard` so that instead of saving values in `@name`, etc, it would be stored directly on, say, `@course`? You could immediately get the benefits of validations and other calculated values, for example, and wouldn't need this extra mapping that has to happen here, and if implemented a certain might even contribute to edit functionality (e.g. a system allows users to pick what they want to edit, and automatically reports on anything not-yet-valid as a guide of what's missing).

Reply:
Agreed – I've introduced a new `NewCourseWizard` model class which encapsulates all the course creation logic.